content/blog/sivayanama-testing-opsramp-apis-with-pytest-fixtures.md

```diff
@@ -0,0 +1,242 @@
+
+
+
+## What Fixtures are 
+
+PyTest fixtures are a special type of python function that provision fixed base line for testing.With the help of this base line we can ensure tests are run in reliable manner and produce consistent results and the same tests can be repeatable.
+
+## Install PyTest
+
+Run the below command 
+
+```shell
+pip install -U pytest
+```
+
+ 
+
+## Verify PyTest installation 
+
+Run the below command 
+
+```shell
+pytest --version
+pytest 7.4.0
+
+```
+
+## Define Fixture
+
+We can define fixtures just by decorating a simple python function with [@pytest.fixture](https://docs.pytest.org/en/6.2.x/reference.html#pytest.fixture) for example 
+
+```python
+import pytest
+
+@pytest.fixture
+def hello_world():
+    return 'Hello, Happy testing'
+```
+
+
+
+## Invoke fixtures
+
+Test functions can invoke fixtures just by declaring the required fixtures as arguments. 
+
+```python
+def test_hello_world(hello_world):
+    assert hello_world == 'Hello, Happy testing'
+```
+
+## Invoke OpsRamp API
+
+Let us try to execute OpsRamp [Get Access Token API](https://develop.opsramp.com/v2/api/auth/tenancy-auth-oauth-token)  as shown below.
+
+```python
+import http.client
+import json
+import traceback
+import urllib
+from asyncio.log import logger
+
+import pytest
+
+def get_auth_header(api_endpoint, client_id, client_secret, grant_type="client_credentials"):
+    """
+    Returns bearer token string in the blow format
+    bearer adfa-afdaf-1599-402c-a3ee-1ed24f597cc8
+    """
+
+    api_connection = http.client.HTTPSConnection(api_endpoint)
+    params = {"client_id": client_id, "client_secret": client_secret, "grant_type": "client_credentials"}
+
+    payload = urllib.parse.urlencode(params)
+
+    headers = {
+        'accept': "application/json",
+        'content-type': "application/x-www-form-urlencoded",
+    }
+
+```
+
+## Define Fixture to get Access Token
+
+You can transform above function as PyTest Fixture by simply decorating the code as shown below 
+
+
+
+@pytest.fixture
+
+def get_auth_header(api_endpoint, client_id, client_secret, grant_type="client_credentials"):
+
+
+
+## Autouse fixtures
+
+Sometimes some of your fixtures are required for all other tests. The above Get acces token is perfect use case for this. Every API call is depending on this API. In this case you can make this fixture as “autouse fixture”  by passing in autouse=True to the fixture’s decorator. The autouse fixure capability will reduce lot of reduanct requests. 
```

Comment:
Suggest to highlight in italic the "_Get Access Token_" 
Suggest to highlight _autouse=True_ in italic
What d you mean with lot of "reduanct" requests. Did you mean redundant request?